mp/src/game/client/momentum/ui/HUD/hud_speedometer_data.cpp

```diff
@@ -0,0 +1,200 @@
+            KeyValues *pSpeedoKV = pGamemodeKV->FindKey(speedoname);
+            if (!pLabel || !pSpeedoKV)
+                continue;
+
+            pLabel->SetVisible(pSpeedoKV->GetBool("visible", false));
+            if (!pLabel->IsVisible())
+                continue;
+
+            int colorize = pSpeedoKV->GetInt("colorize", 0);
+            if (colorize == SPEEDOMETER_COLORIZE_RANGE)
+            {
+                KeyValues *pRangesKV = pSpeedoKV->FindKey("ranges");
+                if (!pRangesKV)
+                    continue;
+
+                RangeList *m_pRangeList = new RangeList();
+                for (KeyValues *pRangeItem = pRangesKV->GetFirstSubKey(); pRangeItem != nullptr;
+                     pRangeItem = pRangeItem->GetNextKey())
+                {
+                    Range_t range;
+                    range.min = pRangeItem->GetInt("min", 0);
+                    range.max = pRangeItem->GetInt("max", 0);
+                    range.color = pRangeItem->GetColor("color");
+                    m_pRangeList->AddToTail(range);
+                }
+                pLabel->SetRangeList(m_pRangeList);
+            }
+            pLabel->SetColorizeType(colorize);
+            pLabel->SetUnitType(pSpeedoKV->GetInt("units", 0));
+        }
+    }
+}
+
+void SpeedometerData::SaveGamemodeData(GameMode_t gametype)
+{
+    if (!g_pFullFileSystem)
+        return;
+
+    KeyValues *pGamemodeKV = m_pGamemodeSetupData->FindKey(g_szGameModes[gametype]);
+    if (pGamemodeKV) // game mode is known
+    {
+        // set speedometer label setups
+        for (KeyValues *pSpeedoItem = pGamemodeKV->GetFirstSubKey(); pSpeedoItem != nullptr;
+             pSpeedoItem = pSpeedoItem->GetNextKey())
+        {
+            const char *speedoname = pSpeedoItem->GetName();
+            SpeedometerLabel *pLabel = GetLabelFromName(speedoname);
+            KeyValues *pSpeedoKV = pGamemodeKV->FindKey(speedoname, true);
+            if (!pLabel || !pSpeedoKV)
+                continue;
+
+            pSpeedoKV->Clear();
+            pSpeedoKV->SetBool("visible", pLabel->IsVisible());
+            pSpeedoKV->SetInt("colorize", pLabel->GetColorizeType());
+            RangeList *pRangeList = pLabel->GetRangeList();
+            if (pRangeList)
+            {
+                KeyValues *pRangesKV = pSpeedoKV->FindKey("ranges", true);
+                pRangesKV->Clear();
+                char tmp[BUFSIZELOCL];
+                for (auto i = 0; i < pRangeList->Count(); i++)
+                {
+                    Range_t *range = &(*pRangeList)[i];
+                    Q_snprintf(tmp, BUFSIZELOCL, "%i", i + 1);
+                    KeyValues *rangeKV = pRangesKV->FindKey(tmp, true);
+                    rangeKV->SetInt("min", range->min);
+                    rangeKV->SetInt("max", range->max);
+                    rangeKV->SetColor("color", range->color);
+                }
+            }
+            pSpeedoKV->SetInt("units", pLabel->GetUnitType());
```

Comment:
This could actually move back into the speedometer label under a `SaveToKV(KeyValues *pOut)` method.

This will keep all the label's settings in the label's class, meaning you could also get rid of `GetRangeList`.